os/fs/smartfs/smartfs_utils.c

```diff
@@ -1771,49 +1767,48 @@ int smartfs_examine_sector(struct smartfs_mountpt_s *fs, char *validsectors, int
 			header = (struct smartfs_chain_header_s *)sectordata;
 			nextsector = SMARTFS_NEXTSECTOR(header);
 
-			if (type == SMARTFS_SECTOR_TYPE_DIR) {
-				if (smart_validatesector(fs->fs_blkdriver, logsector, validsectors) == OK) {
-					(*nusedsectors)++;
-				}
-
+			if (!(node->type == SMARTFS_SECTOR_TYPE_DIR || node->type == SMARTFS_SECTOR_TYPE_FILE)) {
+				ret = -EINVAL;
```

Comment:
EINVAL means bad argument, EINVAL_SECTOR is correct.

Reply:
Changed as EINVAL_SECTOR.